### ISSUE TITLE
Error on batch training

### ISSUE THREAD

**Issue by theFilipko** [charrywhite/elastic_interaction_based_loss#3]
I have used it on the batch in PyTorch and it was throwing an error about wrong dimensions on line 104 in the `elastic_loss.py`. My solution is to change this
`dmn_r = dmn[:,:,:,0] # dmn's real part`
`dmn_i = dmn[:,:,:,1] # dmm's imagine part`
to this
`dmn_r = dmn[:,:,:,:,0] # dmn's real part`
`dmn_i = dmn[:,:,:,:,1] # dmm's imagine part`

This also raises a question: If it is used on batches, should the final loss be divided by the batch size?

**Comment by charrywhite** [comment]
> I have used it on the batch in PyTorch and it was throwing an error about wrong dimensions on line 104 in the `elastic_loss.py`. My solution is to change this
> `dmn_r = dmn[:,:,:,0] # dmn's real part`
> `dmn_i = dmn[:,:,:,1] # dmm's imagine part`
> to this
> `dmn_r = dmn[:,:,:,:,0] # dmn's real part`
> `dmn_i = dmn[:,:,:,:,1] # dmm's imagine part`
> 
> This also raises a question: If it is used on batches, should the final loss be divided by the batch size?

Hi,

1. Original dmn's shape is [B,H,W], its first dim is batchsize.  I guess your modified dmn[:,:,:,:,0]'s second dim is number of channel,  is its shape [B,C,H,W,0]?
2. Yes, it should be.

**Comment by theFilipko** [comment]
Yes, that's correct. Thank you.
